hazardous/metrics/_accuracy_in_time.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+
+from ..utils import check_y_survival
+
+
+def accuracy_in_time(y_test, y_pred, time_grid, quantiles=None, taus=None):
+    r"""Accuracy in time for prognostic models using competing risks.
+
+    .. math::
+
+        \mathrm{acc}(\zeta) = \frac{1}{n_{nc}} \sum_{i=1}^n ~ I\{\hat{y}_i=y_{i,\zeta}\}
+        \overline{I\{\delta_i = 0 \cap t_i \leq \zeta \}}
+
+    where:
+
+    - :math:`\zeta` is a fixed time horizon.
+    - :math:`n_{nc}` is the number of uncensored individuals at :math:`\zeta`.
+    - :math:`\delta_i` is the event experienced by the individual :math:`i` at
+      :math:`t_i`.
+    - :math:`\hat{y} = \text{arg}\max\limits_{k \in [0, K]} \hat{F}_k(\zeta|X=x_i)`
+      where :math:`\hat{F}_0(\zeta|X=x_i) \triangleq \hat{S}(\zeta|X=x_i)`.
+
+      :math:`\hat{y}` is the most probable predicted event for individual :math:`i`
+      at :math:`\zeta`.
+    - :math:`y_{i,\zeta} = \delta_i ~ I\{t_i \leq \zeta \}` is the observed event
+      for individual :math:`i` at :math:`\zeta`.
+
+    The accuracy in time is a metric introduced in [Alberge2024]_ which evaluates
+    whether observed events are predicted as the most likely at given times.
+    This metric measures if the highest predicted event (one of the event of interest
+    or the survival one) corresponds to the one observed at :math:`\zeta` for each
+    patient.
+
+    We remove individuals that were censored at times :math:`t \leq \zeta`, so the
+    accuracy in time essentially represents the accuracy of the estimator on
+    observed events up to :math:`\zeta`.
+
+    In the beginning, every model's accuracy in time will be high because it will
+    predict that the patients have survived, which will be true in most cases. This
+    metric's discriminative power will be for advanced times when the model has to
+    select which event will happen for a given patient.
+
+    While the C-index can help clinicians to priorize treatment allocation by ranking
+    individuals by risk of a given event of interest, the accuracy in time answers
+    a different question: "`what is the most likely event that this individual will
+    experience at some fixed time horizon?`". Conceptually, it helps clinicians choose
+    the right treatment by priorizing the risk for a given individual.
```

Comment:
maybe say that the c-index depends on other individuals in the cohort, while the accuracy in time for an individual does not?